### ISSUE TITLE
Partitioning of stored data

### ISSUE THREAD

**Issue by mikula** [SwedishPensionsAgency/Coldbir#73]
### This is aimed to open a new discussion

Please contribute with ideas before we start to write the code for it

Columns could be partitioned into several files. The current storage could be just a special case with one partition. 
The partitioning could be used in a map/reduce kind of work. A complete data can be stored in different servers but the tasks can be efficient divided among them.
the idea by code example:
- K_all <- db$partition
- k one of 1:length(K_all)
- db["v", 2010, part = k]  <-  rowSums(db["v",c( 2010 , ._), part = k])
  default value of part is 1L
  This meand that all Coldbir data.tables are stored in one partition except cases with explicit paritioning.
- cdb(path, read_only = F, partitioning = v)
  partitioning =1L by default
  otherwise partitioning is a vector of endspositions of each partition

**Comment by mikula** [comment]
Basic operations needed:
1. _index_vector <- db$order(keyColumns = c("var1",..))_
2. _db$permutate(index_vector)_ # permutates  the total database
3. _breaksVect <- db$calcBreaks(nrOfParts = 10)_ # assert that database is sorted
4.  _db$split(breaks = breaksVect)_ # partitioning av the total database
5. (1) + (2) are used in _db$sort(c("var1",..))_
6. (3) + (4) are used in _db$duvide(nrOfParts = 10)_
7. (5) + (6) are used in _db$partitioning(keyColumns = c("var1",...), nrOfParts = 10)_
8. need of a reverse function _db$pullTogether()_ # inverse to _divide_
9. future development _db$addRows(newData)_

**Comment by mikula** [comment]
Rethinking 
Meanwhile I will take another issue.
